src/object/cli_obj.c

```diff
@@ -4389,7 +4389,13 @@ obj_shard_list_prep(daos_obj_list_t *obj_args,
 	int idx;
 	int rc = 0;
 
-	shard_arg->la_nr = (*obj_args->nr + shard_nr - 1) / shard_nr;
+	if (*obj_args->nr < shard_nr) {
+		D_ERROR("Degraded enumeration nr %d > shards %d\n",
```

Comment:
why L4239 is <, and L4393 is > ?

Reply:
ah, error msg is wrong, let me fix it.

Reply:
let me fix it in the following patch, since it is not easy to get all pass these days.